docs/TRANSCRIPTION.md

```diff
@@ -0,0 +1,222 @@
+const outputToken = await handleGenerateToken({ channel: channel, role: 2, uid: outputUid, expireTime: 3600 })
+const url = `https://api.agora.io/v1/projects/${APP_ID}/rtsc/speech-to-text/tasks?builderToken=${builderToken}`
+const body = {
+    "audio": {
+        "subscribeSource": "AGORARTC",
+        "agoraRtcConfig": {
+            "channelName": channel,
+            "uid": botUid,
+            "token": botToken,
+            "channelType": "LIVE_TYPE",
+            "subscribeConfig": {
+                "subscribeMode": "CHANNEL_MODE"
+            }
+        }
+    },
+    "config": {
+        "features": [
+            "RECOGNIZE"
+        ],
+        "recognizeConfig": {
+            "language": "en-US,es-ES",
+            "model": "Model",
+            "output": {
+                "destinations": [
+                    "AgoraRTCDataStream",
+                ],
+                "agoraRTCDataStream": {
+                    "channelName": channel,
+                    "uid": outputUid,
+                    "token": outputToken
+                }
+            }
+        }
+    }
+}
+const res = await makeRequest("POST", url, credential, JSON.stringify(body))
+const data = await res.json()
+const taskId = data.taskId
+
+return sendSuccessfulResponse({
+    taskId: taskId,
+    builderToken: builderToken,
+})
+```
+
+Upon a successful response, you will receive a `taskId`. Both the `taskId` and the `builderToken` will need to be used to stop the transcription, so they must be returned to the caller.
+
+To test this, run your backend using `npm run dev`. You can use cURL to send a `POST` request in your terminal with a body containing `channel`.
+
+```
+curl -X POST http://localhost:4321/api/transcription/start.json \
+  -H "Content-Type: application/json" \
+  -d '{
+    "channel": "test"
+  }'
+```
+
+If your `POST` request is missing information, you will receive a Bad Request response telling you what you missed. If you have a proper request, you will receive a successful response that looks like this:
+
+```
+{"taskId":"<long string>","builderToken":"<another long string>"}
+```
+
+## Stop Transcribing
+To stop the transcription, you must send a `DELETE` command to a URL containing the App ID, Task ID, and the Builder Token.
+
+```ts
+const credential = generateCredential()
+const url = `https://api.agora.io/v1/projects/${APP_ID}/rtsc/speech-to-text/tasks/${taskId}?builderToken=${builderToken}`
+const res = await makeRequest("DELETE", url, credential)
+const data = await res.json()
+
+return sendSuccessfulResponse(data)
+```
+
+To test that it is working, run this cURL command.
+
+```
+curl -X DELETE "http://localhost:4321/api/transcription/stop.json" \
+  -H "Content-Type: application/json" \
+  -d '{
+    "taskId": "<from start command return>",
+    "builderToken": "<from start command return>"
+  }'
+```
+
+You now have a fully working Real Time Transcription backend with Astro.
```

Comment:
This ending feels a bit abrupt.

Reply:
updated to:

With this guide, you have now built a backend that features token generation, cloud recording, and speech to text. All this is within a single Astro project. Pretty neat if you ask me.